### ISSUE TITLE
JS highlight issues after #50293

### ISSUE THREAD

**Issue by alexlamsl** [Microsoft/vscode#51263]
Issue Type: <b>Bug</b>

- open https://github.com/mishoo/UglifyJS2/blob/9e19e63551907be19c8c6b5b75217d7650504436/lib/compress.js
- notice that syntax highlighting starts to go haywire around line 4673

VS Code version: Code 1.24.0 (6a6e02cef0f2122ee1469765b704faf5d0e0d859, 2018-06-06T17:35:40.560Z)
OS version: Windows_NT x64 10.0.14393

Extensions: none

![syntax-colour](https://user-images.githubusercontent.com/1898413/41065107-254039a2-6a10-11e8-8109-b387058287c4.png)

**Comment by mjbvz** [comment]
Duplicate of https://github.com/Microsoft/TypeScript-TmLanguage/issues/625

**Comment by chrmarti** [comment]
@mjbvz I don't see any change on the release branch for this yet. I assume there will eventually be one?

**Comment by mjbvz** [comment]
I’ll pull it in If we have a recovery build with other fixes. Not important enough to force a recovery build on its own imo

**Comment by chrmarti** [comment]
@mjbvz There are a few fixes queued already. You can bring this one in.

**Comment by mjbvz** [comment]
Pulled in with b7557aa2056a1e46ab460a678e0882ad7a834544
